### PR TITLE
[master] Porting #51840 to master

### PR DESCRIPTION
**Pull request by garethgreenaway**
Porting #51840 to master

### CODE REVIEW

**Comment by dwoz** [comment]
@team-core the test failures, docs, and lint all look legit

**Review by twangboy — DISMISSED** [review]

**Comment by dwoz** [comment]
Black needs to be run on this.

File: salt/roster/sshknownhosts.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
+"""
```
Comment (Ch3LL) [inline salt/roster/sshknownhosts.py:1]: versionadded for 3006

**Review by Ch3LL — CHANGES_REQUESTED** [review]
this needs a changelog

**Review by Ch3LL — DISMISSED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**